### ISSUE TITLE
Receiving HTTP Error 500 when trying to capture historical stock data using get.historical()

### ISSUE THREAD

**Issue by shreyprasad** [lukaszbanasiak/yahoo-finance#126]
Hi, 
I am reading a csv file containing the stock ticker symbols of companies in the S&P 500 index and then trying to get the historical data from January 20th to April 30th 2017 (100 days). After 30 or so loops I get the following error: urllib.error.HTTPError: HTTP Error 500: Server Error

I am assuming this is due to rate limiting. I was wondering if there is a way around this? 

here is a snippet of of the loop code: 
for row in df['Symbol']:	
		Stock=Share(row)
		time.sleep(4)	
		History=Stock.get_historical('2017-01-20','2017-04-30')
		time.sleep(4)

Thanks,
Shrey

**Comment by roHyde** [comment]
I get a mix of "HTTP Error 500" and "YQLResponseMalformedError: Response malformed" at random moments. sometimes after 2 results sometimes after 20, at max I have been able to get 40-50 results without error.

**Comment by shreyprasad** [comment]
Hi roHyde, 
I am in the same boat. 
Have you tried to understand what is causing this issue?

**Comment by jkeller1025** [comment]
I haven't been able to pull historical data for over a day now. I believe the package is pulling stock data via YQL console which i believe has a rate limit of 2000 requests / hour for public users.

I've waited and tried to pull info for one stock over a short time window and I still get the same errors mentioned above. Any suggestions or workarounds would be greatly appreciated. Is getting an API key a viable option?

**Comment by madc0ww** [comment]
@jkeller1025 Have you found a workaround? It seems that this critical bug remains for at least 11 days, vide #128 #129 etc...

**Comment by jkeller1025** [comment]
@madc0ww I found a workaround - use Google for the time being. Required some adjustments since the output is slight different than the yahoo-finance package output. Also, the code used in the following link is python 2.# - i tried converting to python 3.# using 2to3 but it still wasn't working. Definitely an inconvenience. Here's a link to the website I used for reference. 

https://trading.cheno.net/downloading-google-finance-historical-data-with-python/
